lib/use_cases/save_items_details.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SaveItemsDetails
+  def initialize(items_gateway:)
+    @items_gateway = items_gateway
+  end
+
+  def execute(items_details:)
+    @items_details = items_details
+
+    errors = []
+    errors.push(missing?)
+    errors = errors.flatten(1)
+
+    return { successful: false, errors: errors } unless errors.empty?
+
+    @items_gateway.save(items_details)
+
+    return { successful: true, errors: [] }
+
+  end
+
+  private
+
+  def missing?
```

Comment:
If it has a '?' does it not have to return a boolean?

Reply:
Yes.